src/AutoMapper/MapperConfiguration.cs

```diff
@@ -201,40 +193,41 @@ public TypeMap ResolveTypeMap(Type sourceType, Type destinationType)
 
         public TypeMap ResolveTypeMap(TypePair typePair)
         {
-            var typeMap = _typeMapPlanCache.GetOrAdd(typePair);
-            if(typeMap != null && typeMap.MapExpression == null && Configuration.CreateMissingTypeMaps)
-            {
-                lock(typeMap)
-                {
-                    typeMap.Seal(this);
-                }
-            }
+            var typeMap = _typeMapPlanCache.GetOrAdd(typePair);
+
+            if (typeMap == null)
+                return null;
+
+            lock (typeMap)

```

Comment:
Oh that lock was already there, it just started blowing up because the type map created could be null.